File: src/main/java/com/pickme/calendar/entity/InterviewDetail.java
package com.pickme.calendar.entity;

import java.time.LocalDateTime;
import java.util.Date;
import java.util.UUID;

import org.springframework.data.mongodb.core.index.Indexed;

import lombok.Builder;
import lombok.Getter;

@Getter
@Builder
public class InterviewDetail {

	@Indexed
	@Builder.Default
	private final String interviewDetailId = UUID.randomUUID().toString();

	@Builder.Default
	private final LocalDateTime createdAt = LocalDateTime.now();

	@Builder.Default
	private LocalDateTime updatedAt = LocalDateTime.now();

	private Company company;

	private Date interviewTime;

	private String position;

	private String category;

	private String description;

	public void touch() {
		this.updatedAt = LocalDateTime.now();
	}

	@Getter
	@Builder
	public static class Company {

		private String name;

		private String location;
	}
}

File: src/main/kotlin/com/pickme/calendar/entity/InterviewDetail.kt
package com.pickme.calendar.entity

import lombok.Builder
import org.springframework.data.mongodb.core.index.Indexed
import java.time.LocalDateTime
import java.util.*

data class InterviewDetail(
    @Indexed
    @Builder.Default
    val interviewDetailId: String = UUID.randomUUID().toString(),

    @Builder.Default
    val createdAt: LocalDateTime = LocalDateTime.now(),

    @Builder.Default
    var updatedAt: LocalDateTime = LocalDateTime.now(),

    val company: Company,

    val interviewTime: Date,

    val position: String,

    val category: String,

    val description: String,
) {
    fun touch() {
        this.updatedAt = LocalDateTime.now()
    }

    data class Company(
        val name: String,

        val location: String,
    )
}
